ipatests/test_integration/test_krbtpolicy.py

```diff
@@ -35,12 +36,19 @@ def maxlife_within_policy(input, maxlife, slush=3600):
 
        Input should be a string like:
        11/19/2019 16:37:40  11/20/2019 16:37:39  krbtgt/...
+       or
+       11/19/19 16:37:40  11/20/19 16:37:39  krbtgt/...
 
        slush defaults to 1 * 60 * 60 matching the jitter window.
     """
     data = input.split()
-    start = datetime.strptime(data[0] + ' ' + data[1], '%m/%d/%Y %H:%M:%S')
-    end = datetime.strptime(data[2] + ' ' + data[3], '%m/%d/%Y %H:%M:%S')
+    if re.match(r'([0-9]{2}\/[0-9]{2}\/[0-9]{4})', data[0]):
+        date_format = "%m/%d/%Y %H:%M:%S"
+    else:
+        date_format = "%m/%d/%y %H:%M:%S"
```

Comment:
Just for arguments sake, if the regex fails and it falls back to two-digit year and that isn't right either for some reason, will the error be observable in the failure logs?

Or should there be two regex and an exception raise if the format doesn't match either of these schemes?

Reply:
- yes, if there is some unexpected value the following ValueError will appear  - for example (three-digit year):
`ValueError: time data '07/28/222 12:50:02' does not match format '%m/%d/%y %H:%M:%S'`

- The same error will occur if  the format for day or month is wrong:
`ValueError: time data '07/58/2022 12:50:02' does not match format '%m/%d/%Y %H:%M:%S'`

- I also thought about adding try/except but IMO the regex  is enough and if there is some unexpected error in the date format from krb5 output we can see ValuError in the log

Reply:
Hello,
do you agree with these changes? Could you merge it? It seems that I do not have write access into freeipa repository.  
Thanks for the help and check.